app/jobs/decidim/action_delegator/export_consultation_results_job.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Decidim
+  module ActionDelegator
+    class ExportConsultationResultsJob < ApplicationJob
+      queue_as :default
+
+      def perform(user, consultation)
+        @consultation = consultation
+
+        export_data = Decidim::Exporters.find_exporter("CSV").new(collection, serializer).export
+        ExportMailer.export(user, "consultation_results", export_data).deliver_now
+      end
+
+      private
+
+      attr_reader :consultation
+
+      def collection
+        consultation.questions.published.includes(:responses).flat_map do |question|
+          Decidim::ActionDelegator::ResponsesByMembership.new(question).query
+        end
```

Comment:
This needs to be reviewed. It's an N+1 as big as Texas which also happens when rendering the view.

Reply:
I think this is causing Heroku to exceed mem quota already in staging :see_no_evil: . I have to say it was already on the edge.